java/bundles/org.eclipse.set.application/src/org/eclipse/set/application/geometry/GeoKanteGeometryServiceImpl.java

```diff
@@ -342,91 +381,145 @@ public List<GEOKanteMetadata> getGeoKanten(final TOP_Kante topKante) {
 	 *            the TOP_Knoten to start from
 	 * @return a list of GEOKanteMetadata for the TOP_Kante
 	 */
-	private List<GEOKanteMetadata> getTOPKanteMetadata(final TOP_Kante topEdge,
+	private List<GEOKanteMetadata> getGeoKanteMetadatas(final TOP_Kante topEdge,
 			final TOP_Knoten start) {
 		final String key = CacheUtils.getCacheKey(topEdge, start);
 
 		return geoKanteMetadas.computeIfAbsent(key, k -> {
 			final List<Bereich_Objekt> bereichObjekt = StreamSupport.stream(
 					getContainer(start).getBereichObjekt().spliterator(), false)
 					.toList();
-			return getTOPKanteMetadata(topEdge, start, bereichObjekt);
+			return getGeoKantenMetadata(topEdge, getGEOKnoten(start),
+					bereichObjekt);
 		});
 	}
 
 	@Override
-	public Pair<GEOKanteCoordinate, BigDecimal> getProjectionCoordinate(
+	public Pair<GEOKanteCoordinate, BigDecimal> getProjectionCoordinateOnTopKante(
 			final Coordinate coor, final TOP_Kante topEdge) {
-		final Pair<GEOKanteMetadata, Coordinate> projectionCoorAndGeoKante = findProjectionCoorAndGeoKante(
-				coor, topEdge);
-		if (projectionCoorAndGeoKante == null) {
+		return getProjectionCoordinate(coor, topEdge);
+
+	}
+
+	private Pair<GEOKanteCoordinate, BigDecimal> getProjectionCoordinate(
+			final Coordinate coordinate, final Basis_Objekt object) {
+		try {
+			final Pair<GEOKanteMetadata, Coordinate> projectionCoorAndGeoKante = getClosestProjection(
+					coordinate, object);
+			if (projectionCoorAndGeoKante == null) {
+				throw new NullPointerException();
+			}
+
+			// Find the GEOKanteSegment, which the projection lie on
+			final GEOKanteMetadata metadata = projectionCoorAndGeoKante
+					.getFirst();
+			final Coordinate projectionCoor = projectionCoorAndGeoKante
+					.getSecond();
+			// Determine topological distance from start of Top_Kante to the
+			// projection
+			final BigDecimal projectionTopDistance = getCoordinateTopDistance(
```

Comment:
@TruongQuangSB We should name it generically here now

```suggestion
			// Determine topological distance from start of object to the
			// projection
			final BigDecimal projectionDistance = getCoordinateTopDistance(
```